build/dependencies.yaml

```diff
@@ -82,21 +82,17 @@ dependencies:
     - path: cluster/images/etcd/migrate/options.go
 
   # Golang
-  - name: "golang"
-    version: 1.15.0-rc.2
+  - name: "golang: upstream version"
+    version: 1.15
     refPaths:
     - path: build/build-image/cross/VERSION
     - path: build/root/WORKSPACE
-      match: (override_)?go_version = "((([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)"
-
-  - name: "golang: upstream container image"
-    version: 1.15rc2
-    refPaths:
+      match: (override_)?go_version = "\d+.\d+(alpha|beta|rc)?\.?\d+"
```

Comment:
I think you want something like:
```
"\d+\.\d+(alpha\d*|beta\d*|rc\d*|\.\d+)?"
```

```
# one or more digits, followed by .
\d+\.
# one or more digits
\d+
# optionally followed by alpha/beta/rc of any number, or a specific patch release
(
  alpha\d*
  |
  beta\d*
  |
  rc\d*
  |
  \.\d+
)?
```

Reply:
follow up for this is fine, just noting that the only reason this currently passes 1.15 is:

```
"\d+.\d+(alpha|beta|rc)?\.?\d+"
"1  .1                     5"
```

and the `.` in the regex would match any character, not just a literal `.`

Reply:
Yep, will issue a follow up to clean up some of the regex in `dependencies.yaml`.
It's a bit Wild Wild West in there at this point.